src/shared/ArtifactPackage.cs

```diff
@@ -68,6 +68,12 @@ static bool FromFilenameOrUrl(string fileName, string url, out ArtifactPackage a
             Architecture = rxGroups["arch"].Value.ToLower();
             IsSnapshot = !rxGroups["snapshot"].Value.IsEmpty();
 
+            // HACK
```

Comment:
Is there a better way to do this, or will this live on forever :)?

Reply:
that's a very good question :)